File: C++/Palindrome_Number.cpp
//? The objective of this C++ Program is to check if a number is palindrome or not.
#include <iostream>
using namespace std;
int main()
{
    int num, A[20], temp = 0, i = 0, j = 0, flag = 0, c = 0, dig = 0, B[20];
    char str_num;
    cout << "Enter a number:";
    cin >> num;
    temp = num;
    while (temp > 0)
    {
        dig = temp % 10;
        A[i] = dig;
        i += 1;
        c++;
        temp /= 10;
    }
    for (i = c - 1, j = 0; i > 0, j < c; i--, j++)
    {
        B[j] = A[i];
    }
    for (i = 0; i < c; i++)
    {
        if (A[i] != B[i])
        {
            flag = 1;
            break;
        }
    }
    if (flag == 1)
    {
        cout << "Not Palindrome!";
    }
    else
    {
        cout << "Palindrome!";
    }
}

File: Java/PalindromeNumber.java

//? The objective of this Java program is to check if a number is palindrome or not.
import java.util.Scanner;

public class PalindromeNumber {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int num;
        int[] a = new int[20];
        int temp = 0;
        int i = 0;
        int j = 0;
        int flag = 0;
        int c = 0;
        int dig = 0;
        int[] b = new int[20];
        System.out.println("Enter a number:");
        num = sc.nextInt();
        temp = num;
        while (temp > 0) {
            dig = temp % 10;
            a[i] = dig;
            i += 1;
            c++;
            temp /= 10;
        }
        i = c - 1;
        j = 0;
        while (i >= -1 || j <= c + 1) {
            if (i == -1) {
                break;
            }

            b[j] = a[i];
            i--;
            j++;
        }
        for (i = 0; i < c; i++) {
            if (a[i] != b[i]) {
                flag = 1;
                break;
            }
        }
        if (flag == 1) {
            System.out.println("Not Palindrome!");
        } else {
            System.out.println("Palindrome!");
        }
        sc.close();
    }
}
